pkg/api/addons.go

```diff
@@ -72,7 +80,7 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultClusterAutoscalerAddonsConfig := KubernetesAddon{
 		Name:    ClusterAutoscalerAddonName,
-		Enabled: to.BoolPtr(DefaultClusterAutoscalerAddonEnabled),
+		Enabled: to.BoolPtr(DefaultClusterAutoscalerAddonEnabled && !cs.Properties.IsAzureStackCloud()),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.